File: frontend/src/types/lead.ts
export enum LeadStatus {
    NEW = "new",
    CONTACTED = "contacted",
    QUALIFIED = "qualified",
    CONVERTED = "converted",
    LOST = "lost"
}

export enum LeadSource {
    WEBSITE = "website",
    REFERRAL = "referral",
    SOCIAL_MEDIA = "social_media",
    DIRECT = "direct"
}

export interface Lead {
    id: number;
    name: string;
    email: string;
    phone: string;
    status: LeadStatus;
    source: LeadSource;
    notes: string;
    assigned_to?: number;
    created_at: string;
    updated_at: string;
}

export interface LeadCreate {
    name: string;
    email: string;
    phone: string;
    status: LeadStatus;
    source: LeadSource;
    notes?: string;
    assigned_to?: number;
}

export interface LeadUpdate {
    name?: string;
    email?: string;
    phone?: string;
    status?: LeadStatus;
    source?: LeadSource;
    notes?: string;
    assigned_to?: number;
}

export interface LeadFilter {
    status?: LeadStatus;
    source?: LeadSource;
    search?: string;
    date_range?: [string, string];
}

export interface LeadStats {
    total_leads: number;
    new_leads: number;
    contacted_leads: number;
    qualified_leads: number;
    converted_leads: number;
    conversion_rate: number;
}

export interface LeadTrend {
    date: string;
    count: number;
}

export interface SourceDistribution {
    source: string;
    count: number;
}

export interface LeadAnalytics {
    lead_trends: LeadTrend[];
    source_distribution: SourceDistribution[];
} 

File: app/schemas/lead.py
from typing import List, Optional, Dict, Any
from pydantic import BaseModel, Field, EmailStr
from datetime import datetime
from enum import Enum
from uuid import UUID

class LeadSource(str, Enum):
    WEBSITE = "website"
    REFERRAL = "referral"
    SOCIAL = "social"
    DIRECT = "direct"
    OTHER = "other"

class LeadStatus(str, Enum):
    NEW = "new"
    CONTACTED = "contacted"
    QUALIFIED = "qualified"
    LOST = "lost"
    CONVERTED = "converted"

class LeadPriority(str, Enum):
    LOW = "low"
    MEDIUM = "medium"
    HIGH = "high"
    URGENT = "urgent"

class LeadBase(BaseModel):
    name: str
    email: Optional[EmailStr] = None
    phone: Optional[str] = None
    source: Optional[LeadSource] = None
    status: Optional[LeadStatus] = LeadStatus.NEW
    notes: Optional[str] = None
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    created_by: Optional[UUID] = None
    updated_by: Optional[UUID] = None

class LeadCreate(LeadBase):
    pass

class LeadUpdate(BaseModel):
    name: Optional[str] = None
    email: Optional[EmailStr] = None
    phone: Optional[str] = None
    source: Optional[LeadSource] = None
    status: Optional[LeadStatus] = None
    notes: Optional[str] = None
    first_name: Optional[str] = None
    last_name: Optional[str] = None
    updated_by: Optional[UUID] = None

class LeadInDBBase(LeadBase):
    id: UUID
    customer_id: UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

class Lead(LeadInDBBase):
    pass

class LeadInDB(LeadInDBBase):
    pass

class LeadList(BaseModel):
    items: List[Lead]
    total: int
    page: int
    size: int
    pages: int

class LeadStats(BaseModel):
    total_leads: int
    active_leads: int
    conversion_rate: float
    average_score: float
    status_distribution: Dict[str, int]
    source_distribution: Dict[str, int]
    priority_distribution: Dict[str, int]
    timeline: List[Dict[str, Any]] 